File: week7/윤여준/가장 큰 수.cpp
#include <bits/stdc++.h>
using namespace std;

bool comp(const string& a, const string& b) {
    return a + b > b + a;
}

string solution(vector<int> numbers) {
    string answer = "";

    vector<string> nums;
    for(int i : numbers) nums.push_back(to_string(i));

    sort(nums.begin(), nums.end(), comp);

    for(string str : nums) answer += str;

    return answer[0] == '0' ? "0" : answer;
}

File: week7/이지훈/가장 큰 수.java
import java.util.*;

public class 가장 큰 수 {
    public String solution(int[] numbers) {

        String[] nums = new String[numbers.length];

        for (int i = 0; i < nums.length; i++) {
            nums[i] = String.valueOf(numbers[i]);
        }

        Arrays.sort(nums, (o1, o2) -> (o2 + o1).compareTo(o1 + o2));
        if(nums[0].equals("0")) return "0";

        StringBuilder sb = new StringBuilder();
        for (String s : nums) {
            sb.append(s);
        }

        return sb.toString();
    }

}